docs/the-contribution-workflow.md

```diff
@@ -0,0 +1,245 @@
+
+If it makes sense, someone from the project will jump in, offer some guidance, and (generally) be pleased with your offer.
+
+# Working on the code
+
+OK, you've got an issue, you've got approval to work on it. Now's the fun part.
+
+Now you need to get Exercism running on your local machine.
+
+Setup can be challenging, but fortunately, Exercism has great docs for [walking you through this process](https://github.com/exercism/exercism.io/blob/master/docs/setting-up-local-development.md).
+
+If you can work through the steps in the above guide, great! If you're still not positive what to do, read on.
+
+1. Fork the Exercism repository to your github account.
+
+The URL for the repo looks like https://github.com/exercism/exercism.io
+
+You want it to be https://github.com/YOUR-USERNAME-HERE/exercism.io
+
+"[Forking](https://help.github.com/articles/fork-a-repo/)" is how we do that.
+
+
+
+![Forking the Repo](/docs/img/beginner_contribution_fork_repo.gif)
+
+Go to https://github.com/exercism/exercism.io, and click "Fork". When it asks "Where should we fork this repository", choose your user account.
+
+2. Clone down the forked repo
+
+Now, from the repo in your account (the URL should be something like https://github.com/YOUR-USERNAME-HERE/exercism.io)
+
+Click "clone or download" and get the URL for cloning it.
+
+In your terminal, run
+
+```shell
+git clone <url from github here>
+```
+
+It might look something like this:
+
+![Cloning down the repo](/docs/img/beginner_contribution_clone_repo.gif)
+
+Once it finishes downloading, change into the new directory, and congrats! You've got Exercism cloned down to your machine!
+
+
+3. Track the original Exercism repo
+
+As you might expect, the Exercism codebase changes constantly. It's important, therefore, to make sure you can easily stay up-to-date with new changes.
+
+To do that, you have to add the original Exercism repo as one of your remotes.
+
+To show what will happen, fell free to run `git remote -v` in your terminal.
+
+It should show something like this:
+
+```shell
+$ git remote -v
+origin	git@github.com:josh-works/exercism.io.git (fetch)
+origin	git@github.com:josh-works/exercism.io.git (push)
+```
+
+So, we want to add another "remote", pointing to https://github.com/exercism/exercism.io.git
+
+To do that, run:
+
+`git remote add upstream https://github.com/exercism/exercism.io.git`
+
+Now, when you run `git remote -v`, you should have _two_ remotes:
+
+```shell
+$ git remote -v
+
+origin	git@github.com:josh-works/exercism.io.git (fetch)
+origin	git@github.com:josh-works/exercism.io.git (push)
+upstream	https://github.com/exercism/exercism.io.git (fetch)
+upstream	https://github.com/exercism/exercism.io.git (push)
+```
+
+This is perfect. Now, go update your current repo with the 'upstream' repo, just run
+
+```shell
+git pull upstream master
```

Comment:
oh, Docker... can you link me to a Docker guide? I've never actually used it, so it *feels* like it's more work than just forking/cloning a repo, but that's just my perception.

Right on w/the git commands, too. 👍